build.gradle

```diff
@@ -22,23 +28,32 @@ targetCompatibility = 1.8
 repositories {
     mavenCentral()
     maven {
-        url "http://artifactory.terasology.org/artifactory/virtual-repo-live"
+        def repoViaEnv = System.getenv()["RESOLUTION_REPO"]
+        if (rootProject.hasProperty("alternativeResolutionRepo")) {
+            // If the user supplies an alternative repo via gradle.properties then use that
+            name "from alternativeResolutionRepo property"
+            url alternativeResolutionRepo
+        } else if (repoViaEnv != null && repoViaEnv != "") {
+            name "from \$RESOLUTION_REPO"
+            url = repoViaEnv
+        } else {
```

Comment:
Why is this necessary, or what is it good for?

Reply:
I recognize this code, along with the `publishing` stuff below, from Terasology's publish.gradle.

It's the stuff I'm always afraid of breaking because it's not part of a workflow I use so I don't know how to test it or how much of it is covered by CI.

Reply:
This is the scaffolding that lets the extra magic in `gradle.properties` work for Terasology, yep. It lets you do the stuff like:

```
alternativeResolutionRepo=http://artifactory.terasology.org/artifactory/virtual-nanoware-and-remote
publishRepo=nanoware-snapshot-local`
```

And you'll be resolving + publishing artifacts against the Nanoware repos in Artifactory, excluding anything that's _live_ in the regular Terasology repos.

Jenkins uses the same approach for Nanoware fork builds, they all get those values via env vars added on the [Nanoware _folder_ in Jenkins](http://jenkins.terasology.io/teraorg/job/Nanoware/configure), without needing it in a `gradle.properties` at all.

It is kinda overkill in here, but did let me verify via the Nanoware snapshot repo that a publish from this project looked right.

It is in essence entirely outside the normal functionality, unless those props or env vars are set it does nothing and goes with the defaults.